types/accedo__accedo-one/index.d.ts

```diff
@@ -0,0 +1,405 @@
+// Type definitions for @accedo/accedo-one 4.0
+// Project: https://www.accedo.tv/one
+// Definitions by: Alexander P. Cerutti <https://github.com/alexandercerutti>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.1
+
+/**
+ * @example
+ * ~ import accedoOne from "@accedo/accedo-one"
+ * ~ const accedo = accedoOne({ ... });
+ * ~
+ * ~ accedo.getEntries(...);
+ *
+ * @param config
+ */
+
+export as namespace accedoOne;
+export default function accedoOne(config: AccedoConfig): AccedoClient;
```

Comment:
This isn’t an ES module; there’s no default export:

```suggestion
declare function accedoOne(config: AccedoConfig): AccedoClient;
export = accedoOne;
```

Reply:
I'm okay with the change to be done, but I'm not quite sure. I mean, the documentation says it can be imported also with ES Syntax, so I think it could be... or must the file end in `.mjs` to be an ES module?

https://github.com/Accedo-Products/accedo-one-sdk-js

Reply:
Judging by the [actual source code](https://unpkg.com/@accedo/accedo-one@4.0.2/dist/accedo-one.umd.min.js), this is a UMD bundle. To be a default export in a UMD module, it would need an `__esModule` marker and to actually set `exports.default`. You can see what that looks like here: https://www.typescriptlang.org/play/?target=1&module=3#code/KYDwDg9gTgLgBAE2AMwIYFcA29noHYDGMAlhHgBQCUcA3gL5A

You _can_ import a UMD export or a CJS export with an ES6-style default import if you have some interop code, which is often provided by Babel, bundlers, and/or TypeScript with `--esModuleInterop`. So the docs aren’t exactly _wrong_, they just don’t tell you anything about what the _export_ shape looks like.

Reply:
Oh wow! I would never have expected this. This is really interesting, thank you for the explanation. I knew that it was a UMD bundle, but still didn't know it wasn't ES Module.

Yes, they only say that the package exports an `accedoOne` factory function, but actually they should detail more about how you can import it through different systems. You mean this, didn't you?

Isn't this something pretty common with packages? I've never seen a package that actually fully explains the export shape (probably because many are made only for CommonJS?).

Reply:
Module formats are incredibly confusing and highly misunderstood, but for most users it doesn’t really matter, because ES interop code is easy to come by, so for end users, stuff just works. But the module authors (or at least, the authors of the tools the pack source code) have to understand that confusing landscape, and that includes those of us writing type definitions for them. There’s an overview in [our handbook](https://www.typescriptlang.org/docs/handbook/modules.html#guidance-for-structuring-modules), but I agree it’s confusing.

Reply:
I agree. In fact for the Typescript code I'm using at work (which gets bundled through Webpack), I'm using everything with `--esModuleInterop`, without really understanding what it was doing.